### PR TITLE
Fix fork state after recovery is successful

### PR DESCRIPTION
**Pull request by richardhuaaa**
This fixes an issue found via @cameronvoell's validation: after a forked client successfully recovers from their forked state, they seem to remain evaluating themselves as forked, and continue to issue re-add requests with each worker iteration after that.

There are two fixes here:

1. After a readd welcome is received, we need to update not only the `readd_status`, but also the `is_forked` flag. This fixes the bulk of the issue.
2. When checking for forks, we need to ignore local commit log entries from *before* the latest Welcome was applied. This fixes any edge cases around stale data after the group has been reinitialized.

Also included an end-to-end unit test that validates both of these fixes (fails without either one of the fixes).

(1) was not previously mentioned in the XIP, also updated it to include this.

### CODE REVIEW

**Comment by richardhuaaa** [comment]
* **#2706** <a href="https://app.graphite.com/github/pr/xmtp/libxmtp/2706?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/xmtp/libxmtp/2706?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>mergequeue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/xmtp?ref=merge-queue-instructions-in-stack-comment&prId=2706).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: xmtp_db/src/encrypted_store/local_commit_log.rs [hunk 1]
```diff
@@ -213,17 +214,27 @@ impl<C: ConnectionExt> QueryLocalCommitLog for DbConnection<C> {
...
-        let after_cursor = after_cursor as i32;
+        let latest_welcome_rowid: Option<i32> = self.raw_query_read(|db| {
+            dsl::local_commit_log
+                .select(diesel::dsl::max(dsl::rowid))
+                .filter(dsl::group_id.eq(group_id))
+                .filter(dsl::rowid.gt(after_cursor as i32))
+                .filter(dsl::commit_type.eq(Some("Welcome".to_string())))
+                .first(db)
+        })?;
+        let gte_cursor = latest_welcome_rowid.unwrap_or(after_cursor as i32 + 1);
```
Comment (tylerhawkes) [inline xmtp_db/src/encrypted_store/local_commit_log.rs:225]: this could use saturating add even though it's unlikely to ever get that far.
